tests/ui/on-unimplemented/multiple-impls.stderr

```diff
@@ -63,9 +60,8 @@ error[E0277]: the trait bound `[i32]: Index<Bar<u32>>` is not satisfied
   --> $DIR/multiple-impls.rs:39:5
    |
 LL |     Index::index(&[] as &[i32], Bar(2u32));
-   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ on impl for Bar
```

Comment:
Can you remove the `rustc_on_unimplemented` from impls in these test files?

Reply:
ideally we'd make it a validation error or something to put this on an impl

Reply:
> so I plan on removing it in the future.

why not now? 🤔

Reply:
for a couple reasons:
- in https://github.com/rust-lang/rust/pull/139141/commits/b1bc7255bb16f2500d5bdcbc17c06c2acd91c7f0 I deleted its only use in std, but it's still present in the beta std. I'm unsure whether raising an error would cause problems.
- rather than adding that check to this PR I'd rather do it in a future hir attributes related PR
- I'm not intimately familiar with this code so didn't feel like raising new errors.

I'm happy to start raising errors now if you'd prefer.

Reply:
Well, I'd at least prefer for us to clean up the UI tests so they don't look like they're buggy (i.e. have annotations on impls, but don't use them). 

I feel like the easiest way of doing that is just making `rustc_on_unimplemented` an error then clean up the new errors in the UI tests, but it's up to you I guess.

Reply:
the "now the ui tests look buggy" argument is compelling to me, i'll experiment with a hard error tomorrow

Reply:
> ideally we'd make it a validation error or something to put this on an impl

it's not a validation error to put it anywhere, you can put it on structs, functions, etc. I would like to start erroring on all of that after migrating the attribute itself.